Find a functionally equivalent code:android.security.KeyChain.bind:COMMENT
Method Modifier: public      static      hidden      
Comment:/**
 * @hide for reuse by CertInstaller and Settings.
 *
 * Caller should call unbindService on the result when finished.
 */

@@ -1,35 +1,3 @@
 {
-    if (context == null) {
-        throw new NullPointerException("context == null");
-    }
-    ensureNotOnMainThread(context);
-    final BlockingQueue<IKeyChainService> q = new LinkedBlockingQueue<IKeyChainService>(1);
-    ServiceConnection keyChainServiceConnection = new ServiceConnection() {
-
-        volatile boolean mConnectedAtLeastOnce = false;
-
-        @Override
-        public void onServiceConnected(ComponentName name, IBinder service) {
-            if (!mConnectedAtLeastOnce) {
-                mConnectedAtLeastOnce = true;
-                try {
-                    q.put(IKeyChainService.Stub.asInterface(service));
-                } catch (InterruptedException e) {
-                // will never happen, since the queue starts with one available slot
-                }
-            }
-        }
-
-        @Override
-        public void onServiceDisconnected(ComponentName name) {
-        }
-    };
-    Intent intent = new Intent(IKeyChainService.class.getName());
-    ComponentName comp = intent.resolveSystemService(context.getPackageManager(), 0);
-    intent.setComponent(comp);
-    boolean isBound = context.bindService(intent, keyChainServiceConnection, Context.BIND_AUTO_CREATE);
-    if (!isBound) {
-        throw new AssertionError("could not bind to KeyChainService");
-    }
-    return new KeyChainConnection(context, keyChainServiceConnection, q.take());
+    return bindAsUser(context, Process.myUserHandle());
 }

